core/src/main/java/org/veriblock/sdk/rewards/PopRewardCalculator.java

```diff
@@ -144,16 +144,11 @@ public static int getBestPublicationHeight(List<AltPublication> publications) th
     }
 
     public static BigDecimal calculatePopDifficultyForBlock(List<AltChainBlock> blocksInterval) throws SQLException {
-        if(blocksInterval.size() != config.popRewardSettlementInterval + config.popDifficultyAveragingInterval) {
-            throw new IllegalArgumentException("The amount of blocks must be equal to popRewardSettlementInterval + popDifficultyAveragingInterval");
-        }
-
         BigDecimal difficulty = BigDecimal.ZERO;
 
         Collections.sort(blocksInterval);  // make the ascending order for the blocks in the collection, it needs for the correct calculation of the pop score
 
-        for(int i = 0; i < config.popDifficultyAveragingInterval; ++i)
-        {
+        for(int i = 0; i < blocksInterval.size() - config.popRewardSettlementInterval; ++i)  {
```

Comment:
Before this change it was obvious that we are calculating average score. Now you have some obscure blocks interval. I think your change should be reverted.

Reply:
This change allows to return the value even if we dont have the whole interval and we can get pop rewards exactly after the payment delay. Without this changes we should wait the block height is payment_delay + difficulty_averaging_interval (in case when pop mining starts from the genesis block of the altchain).